### DISCUSSION TITLE
Yolov5 working slow on nvidia jetson nano 2gb

### DISCUSSION THREAD

**Discussion by domcio01** [ultralytics/yolov5#6651] [Q&A] [answered]
Hi :) i'm trying to run yolov5 on nvidia jetson nano 2gb with different weights but it runs very slow (30 sec before even fusing layers and about 2-3 minutes before it starts detecting ) is there any thing i can do so it works fluently ? i need it to work with CSI camera with at least 20 fps.

**Comment by glenn-jocher** [comment, marked as answer]
👋 Hello! Thanks for asking about **inference speed issues**. YOLOv5 🚀 can be run on CPU (i.e. `--device cpu`, slow) or GPU if available (i.e. `--device 0`, faster). You can determine your inference device by viewing the YOLOv5 console output:

### detect.py inference
```bash
python detect.py --weights yolov5s.pt --img 640 --conf 0.25 --source data/images/
```
<img width="944" alt="detect.py" src="https://user-images.githubusercontent.com/26833433/140100842-e130b5f7-64cb-46c9-9386-72abadd19828.png">

### YOLOv5 PyTorch [Hub](https://github.com/ultralytics/yolov5/issues/36) inference
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')

# Images
dir = 'https://ultralytics.com/images/'
imgs = [dir + f for f in ('zidane.jpg', 'bus.jpg')]  # batch of images

# Inference
results = model(imgs)
results.print()  # or .show(), .save()
# Speed: 631.5ms pre-process, 19.2ms inference, 1.6ms NMS per image at shape (2, 3, 640, 640)
```

### Increase Speeds

If you would like to increase your inference speed some options are:

- Use batched inference with [YOLOv5 PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)
- Reduce `--img-size`, i.e. 1280 -> 640 -> 320
- Reduce model size, i.e. YOLOv5x -> YOLOv5l -> YOLOv5m -> YOLOv5s -> YOLOv5n
- Use half precision FP16 inference with `python detect.py --half` and `python val.py --half`
- Use a faster GPUs, i.e.: P100 -> V100 -> A100
- [Export](https://github.com/ultralytics/yolov5/issues/251) to ONNX for up to 3x CPU speedup
- [Export](https://github.com/ultralytics/yolov5/issues/251) to TensorRT for up to 5x GPU speedup
- Use a free GPU backends with up to 16GB of CUDA memory: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Good luck 🍀 and let us know if you have any other questions!
